game-app/game-core/src/main/java/games/strategy/engine/ClientFileSystemHelper.java

```diff
@@ -38,19 +40,32 @@ public static Path getRootFolder() {
     }
   }
 
-  private static Path getCodeSourceFolder() throws IOException {
-    final ApplicationContext applicationContext = Services.loadAny(ApplicationContext.class);
-    final @Nullable CodeSource codeSource =
-        applicationContext.getMainClass().getProtectionDomain().getCodeSource();
-    if (codeSource == null) {
-      throw new IOException("code source is not available");
-    }
+  public static void setCodeSourceFolder(final Path sourceFolder) {
+    codeSourceLocation = sourceFolder;
+  }
```

Comment:
It's invoked in my automated UI testing which I didn't commit because  
a) I wouldn't know where to put teh test file, because it cannot be included in the automated test run as you explained to me
b) it's written in Kotlin (which was way more fun and IMHO resulted in much better code), and I didn't want to bother you with a programming language discussion on top of what you already have to do regarding QS of my pull request.

Removing it would make it a bit more difficult for me to continue contributing, but I'm not sure how much more I'd like to do (other than are one or two small bug fixes already waiting for this PR to be done).
According to my research having a code source folder that depends on the current working directory can be considered a design weakness, so maybe ```setCodeSourceFolder``` should be used, even if it isn't right now. 
Maybe the whole resource loading should be refactored - again something where I probably would cause more trouble than help, so I refrained from it. If ```setCodeSourceFolder``` would survive that refactoring is something I don't know.

Reply:
I'm working on updating resource loader so we can get the skin switching functionality back. If anything, resource loader has become a mess.

The update here to add a custom test tie-in seems pretty minimal. It is generally best to delete these, the inconvenience to one is usually outweighed by the long term maintenance cost incurred by the many. On another point there is a slippery slope argument, if extra code is added to support non-committed code, where is the line? What if more changes were needed, when would we say no?

Reply:
The next commit will contain automated tests that require ```setCodeSourceFolder```. So I'll leave it in the code.